File: largestprime.py
import PIL
import math
# Problem 3 Largest Prime Factor
def lpf(number):
    factor = 2
    while (number > factor):
        if(number % factor == 0):
            number = number/factor
            factor = 2
        else:
            factor += 1
    return factor


out = lpf(600851475143)
print(out)

File: largestprime.c
#include <stdio.h>
#include <stdlib.h>

long lpf(long num){
    long factor = 2;
    while(num > factor){
        if(num % factor == 0){
            num = num / factor;
            factor = 2;
        }
        else{
            factor ++;
        }
    }
    return factor;
}

int main(){
	printf("%d\n", lpf(600851475143));
	return 0;
}
